viz/cmd/profile.go

```diff
@@ -73,6 +73,34 @@ func newCmdProfile() *cobra.Command {
   linkerd viz profile -n emojivoto web-svc --tap deploy/web --tap-duration 10s --tap-route-limit 5
 `,
 		Args: cobra.ExactArgs(1),
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			// Skip providing suggestions if one or more arguments are provided
+			// We either have a suggestion selected or more multiple args are provided
+			// which is not allowed for this command.
+			if len(args) > 0 {
+				return nil, cobra.ShellCompDirectiveError
+			}
+
+			k8sAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+			if err != nil {
+				return nil, cobra.ShellCompDirectiveError
+			}
+
+			if options.namespace == "" {
+				options.namespace = pkgcmd.GetDefaultNamespace(kubeconfigPath, kubeContext)
+			}
+
+			cc := k8s.NewCommandCompletion(k8sAPI, options.namespace)
+
+			// Profiles require completion for only services so prepend the service resource
+			// type to the list of args
+			results, err := cc.Complete([]string{"service"}, toComplete)
```

Comment:
nit: do we need the plural form of the resource? I see for `ConfigureNamespaceFlagCompletion` we use "namespaces", in the grand scheme of things it doesn't seem to matter (as the completion still works) but wondering if it should be changed just for consistency's sake?

lmk what you think.

Reply:
Agreed we should make this consistent. It'd be even better to use the resource constants available in the `k8s.go` so we don't have to decide on using singular or plural resource type names.